firmware/gpio/gpio.cpp

```diff
@@ -0,0 +1,157 @@
+/**************************
+
+                              gpio driver
+
+***************************/
+#pragma once
+
+#include "LPC17xx.h"
+#include <cstdint>
+#include <iostream>
+#include <string>
+using namespace std;
+
+class Gpio{
+    public:
+    int port_,pin_;
```

Comment:
Why do the members need to be public? These can be made private.

Reply:
Did not make the members private.